processor/src/operations/decorators/mod.rs

```diff
@@ -7,9 +11,79 @@ impl Process {
     // DEBUGGING
     // --------------------------------------------------------------------------------------------
 
-    /// TODO: implement
-    pub fn op_debug(&self) -> Result<(), ExecutionError> {
-        unimplemented!();
+    /// Prints out debugging information based on options passed.
+    pub fn op_debug(&mut self, options: DebugOptions) -> Result<(), ExecutionError> {
+        match options {
+            DebugOptions::All => {
+                self.print_stack(None);
+                self.print_mem(None, None);
+            }
+            DebugOptions::Stack(n) => self.print_stack(n),
+            DebugOptions::Mem(n, m) => self.print_mem(n, m),
+            DebugOptions::Local(_) => self.print_local(),
+        }
+
+        Ok(())
+    }
+
+    // DEBUG HELPER
+    // ---------------------------------------------------------------------------------
+
+    /// Prints stack information for debugging.
+    /// If n is passed, this prints the top n items in the stack.
+    fn print_stack(&self, n: Option<usize>) {
+        let states = self.stack.get_values(n);
+        for (i, value) in states.iter().enumerate() {
+            info!("stack [ index: {:?}, value: {:?} ]", i, value.as_int());
+        }
+    }
+
+    /// Create the display string for printing a word.
+    /// Prints <empty> if no value exists.
+    fn fmt_word(word: Option<Word>) -> String {
+        match word {
+            Some(v) => format!(
+                "[{}]",
+                v.iter()
+                    .map(|v| v.as_int().to_string())
+                    .collect::<Vec<String>>()
+                    .join(", ")
+            ),
+            None => String::from("<empty>"),
+        }
+    }
+
+    /// Print the local variable that fmp pointer is referring to in memory.
+    fn print_local(&self) {
+        let local = self.memory.get_value(self.system.fmp().as_int());
+        info!("local [ {} ]", Process::fmt_word(local));
+    }
+
+    /// Print memory with optional starting and ending addresses.
+    fn print_mem(&self, n: Option<u64>, m: Option<u64>) {
```

Comment:
Would be great to print out total number of memory used somewhere in memory section.